server/src/index.js

```diff
@@ -26,31 +26,31 @@ const main = async () => {
     if (data.type === 'news') {
       if (!await db.collection('Data').findOne({ url: data.url })) {
         const r = await db.collection('Data').insertOne(data)
-        console.log('News added to the database.')
-        res.status(201).json({ success: true })
+        const id = r.insertedId.toHexString()
+        res.status(201).send({ post_id: id })
```

Comment:
You use "data" and now "posts". You should be consistent. Let's avoid data since it is too generic and use posts.